### PR TITLE
New testing api

### PR DESCRIPTION
**Pull request by quincyle**
# Semver

**This project uses [semver](http://semver.org), please check the scope of this PR:**

- [ ] #none#
- [ ] #patch#
- [ ] #minor#
- [X] #major#

# CHANGELOG

* **Updated** tests to new Ember testing API.
* **Updated** exported helpers to align with new Ember testing API.

### CODE REVIEW

File: addon-test-support/setup-test.js [hunk 1]
```diff
@@ -3,16 +3,95 @@
...
+ * @returns {Test} test context object with a subject method
+ */
+export function createTestContextForSerializer (name) {
+  return {
+    subject (options = {}, nameOverrides) {
+      if (!this.owner) {
+        console.error('Make sure you have run setup() before trying to access subject.')
+      }
+
+      this.owner.lookup('service:store').createRecord(nameOverrides || name, options)
```
Comment (EWhite613) [inline addon-test-support/setup-test.js:69]: Do we need this?

File: addon-test-support/setup-test.js [hunk 2]
```diff
@@ -3,16 +3,95 @@
...
+ * @returns {Test} test context object with a subject method
+ */
+export function createTestContextForAdapter (name) {
+  return {
+    subject (options = {}, nameOverrides) {
+      if (!this.owner) {
+        console.error('Make sure you have run setup() before trying to access subject.')
+      }
+
+      this.owner.lookup('service:store').createRecord(nameOverrides || name, options)
```
Comment (EWhite613) [inline addon-test-support/setup-test.js:88]: Same comment

File: tests/test-helper.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
-import resolver from './helpers/resolver'
-import {setResolver} from 'ember-mocha'
+// import resolver from './helpers/resolver'
+// import {setResolver} from 'ember-mocha'
```
Comment (EWhite613) [inline tests/test-helper.js]: remove commented out stuff

File: addon-test-support/setup-test.js (hunk 1 above)
Comment (quincyle) [reply addon-test-support/setup-test.js:69]: Yup, testing methods on Serializer and Adapter will easily trigger a record/model lookup under the hook. And the test will error out when there's no corresponding record.

**Review by EWhite613 — APPROVED** [review]

**Pull request merged**